adapters/platform_win32.c

```diff
@@ -96,21 +155,43 @@ STRING_HANDLE platform_get_platform_info(void)
         DWORD product_type;
         if (GetProductInfo(osvi.dwMajorVersion, osvi.dwMinorVersion, 0, 0, &product_type))
         {
-            result = STRING_construct_sprintf("(native; WindowsProduct:0x%08x %d.%d; %s)", product_type, osvi.dwMajorVersion, osvi.dwMinorVersion, arch);
+            result = STRING_construct_sprintf("(native; WindowsProduct:0x%08x %d.%d; %s", product_type, osvi.dwMajorVersion, osvi.dwMinorVersion, arch);
         }
     }
 
     if (result == NULL)
     {
         DWORD dwVersion = GetVersion();
-        result = STRING_construct_sprintf("(native; WindowsProduct:Windows NT %d.%d; %s)", LOBYTE(LOWORD(dwVersion)), HIBYTE(LOWORD(dwVersion)), arch);
+        result = STRING_construct_sprintf("(native; WindowsProduct:Windows NT %d.%d; %s", LOBYTE(LOWORD(dwVersion)), HIBYTE(LOWORD(dwVersion)), arch);
     }
 #pragma warning(default:4996)
 
     if (result == NULL)
     {
         LogError("STRING_construct_sprintf failed");
     }
+
+    if (options & PLATFORM_OPTION_RETRIEVE_SQM)
```

Comment:
If STRING_construct has failed, this block should not be executed. The if-else blocks need to be rearranged to avoid that.